### PR TITLE
Question button functionality

### LINKED ISSUE #54: Next/Back button functionality
- create functionality with all conditional logic of the next and back buttons in Question component.

### PR DESCRIPTION
**Pull request by timkeresey**
### What does this PR do?

- Implements functionality for "next" and "back" buttons on question component.
- Implements error handling for form requirements.
- Updates global context with form answers.

### Where should the reviewer start?

src/Question/Question.jsx

### How should this be manually tested?

src/Question/Question.test.jsx

### Any background context you want to provide?

- A "required" attribute will 

### What are the relevant tickets?

closes #54 

### Questions:

none

### CODE REVIEW

File: src/components/Question/Question.tsx [hunk 1]
```diff
@@ -1,36 +1,67 @@
...
+    //     setError(true);
+    //     setErrorMessage('Sorry but we need this information')
+    //     setTimeout(()=> setError(false), 2500)
+    //   } else if(index < questionSet.length && answerInput[questionSet[index]]) {
+    //     setIndex(prevIndex => prevIndex + 1);
+    //     setError(false)
+    //   }
+    // }
+
+    const nextQuestion = () => {
```
Comment (BlakeDonn) [inline src/components/Question/Question.tsx:37]: still wondering how we can move these set states out of nextQuestion. I think a custom hook may make the most sense if we have time

**Review by BlakeDonn — APPROVED** [review]
Looks good to me!

File: src/components/Question/Question.tsx (hunk 1 above)
Comment (Atos20) [reply src/components/Question/Question.tsx:37]: I completely agree

File: src/components/Question/Question.tsx [hunk 2]
```diff
@@ -1,36 +1,67 @@
...
-    const [answerInput, updateAnswer] = useState<any>({question: '', answer: ''});
+    const questionSet = Object.keys(questionContext);
 
-    let currentAnswer = Object.keys(answerContext).find(question => answerContext[question])
-    let theQuestion = questionContext[currentQuestion]
+    //local state
+    const [answerInput, updateAnswer] = useState<any>({});
+    const [index, setIndex] = useState<number>(0);
+    // const[error, setError] = useState<boolean>(false);
+    const [errorMessage, setErrorMessage] = useState<string>('');
```
Comment (Atos20) [inline src/components/Question/Question.tsx:21]: Do we want to keep `error` or `errorMessage` ?

File: src/components/Question/Question.css [hunk 3]
```diff
@@ -67,10 +67,37 @@
...
 .buttons-box{
-  grid-area: 6/ 4/ 7 / 8;
+  grid-area: 5/ 4/ 7 / 8;
   display: inline-flex;
   justify-content: space-between;
   margin: 4%;
+  position: relative;
+}
+
+.bx{
```
Comment (Atos20) [inline src/components/Question/Question.css:77]: this code is what solves the issue when the back button doesn't display on the DOM.

File: .eslintcache [hunk 4]
```diff
@@ -1 +1 @@
...
..."line":13,"column":11,"nodeType":"71","messageId":"72","endLine":13,"endColumn":24},{"ruleId":"69","severity":1,"message":"73","line":1,"column":17,"nodeType":"71","messageId":"72","endLine":1,"endColumn":25},{"ruleId":"69","severity":1,"message":"74","line":1,"column":27,"nodeType":"71","messageId":"72","endLine":1,"endColumn":36},{"ruleId":"69","severity":1,"message":"73","line":1,"column":16,"nodeType":"71","messageId":"72","endLine":1,"endColumn":24},{"ruleId":"75","severity":1,"message":"76","line":7,"column":7,"nodeType":"77","endLine":7,"endColumn":85},"@typescript-eslint/no-unused-vars","'Link' is defined but never used.","Identifier","unusedVar","'useState' is defined but never used.","'useEffect' is defined but never used.","jsx-a11y/anchor-is-valid","The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md","JSXOpeningElement","'answerContext' is assigned a value but never used."]
```
Comment (Atos20) [inline .eslintcache:1]: This file depresses me. 💯

**Review by Atos20 — APPROVED** [review]

**Pull request merged**